tool/sauce_runner.dart

```diff
@@ -0,0 +1,247 @@
+  import 'package:test/test.dart';
+  Future waitForCondition(bool condition(),
+    {Duration interval: const Duration(milliseconds: 5),
+    Duration timeout: const Duration(seconds: 5)}) {
+    Completer c = new Completer();
+    DateTime now = new DateTime.now();
+    new Timer.periodic(interval, (timer) {
+      if (condition()) {
+        timer.cancel();
+        c.complete();
+      }
+      if (new DateTime.now().difference(now) > timeout) {
+        timer.cancel();
+        throw ('Conditional check did not succeed within \${timeout}');
+      }
+    });
+    return c.future;
+  }
+  main() async {
+    Stopwatch time = new Stopwatch();
+    await time.start();
+    String innerHtml;
+    await new Future.delayed(new Duration(milliseconds:500));
+    runner.main();
+    await waitForCondition(() {
+      List<Element> elementList = document.querySelectorAll('div');
+      innerHtml = elementList.last.innerHtml;
+      return innerHtml.contains('passed') || innerHtml.isEmpty;
+    }, timeout: new Duration(seconds: $timeForTestCompletion));
+    List<Element> elementList = document.querySelectorAll('div');
+    await time.stop();
+    RegExp exp = new RegExp(r'[\\+](\\d+)[^-]+[\\-]?(\\d+)?');
+    if (innerHtml.contains('passed')) {
+      Match match = exp.firstMatch(elementList.last.innerHtml);
+      js.context['global_test_results'] = new js.JsObject.jsify(
+            {'passed': match.group(1), 'failed': 0, 'total': match.group(1), 'duration': time.elapsed.inMilliseconds, 'tests':[]});
+      } else {
+      Match match = exp.firstMatch(elementList[elementList.length-2].innerHtml);
+      int total = int.parse(match.group(1)) + int.parse(match.group(2));
+      js.context['global_test_results'] = new js.JsObject.jsify(
+            {'passed': match.group(1), 'failed': match.group(2), 'total': total.toString(), 'duration': time.elapsed.inMilliseconds, 'tests':[]});
+      }
+    }
+  """;
+
+// verify that a 200 response is returned from sauce and then
+// return the response
+Map sauceRequestHandling(String response) {
+  RegExp exp = new RegExp(r'(.*)"([\d]*)"$');
+  Match match = exp.firstMatch(response);
+  if (!match.group(2).contains('200')) {
+    String errorMsg = '\nResponse stats: ' + match.group(2);
+    errorMsg += '\n' + match.group(1);
+    throw new Exception(errorMsg);
+  }
+  return JSON.decode(match.group(1));
+}
+
+Future executeSauceTests(Process pubProcess) async {
+  File htmlTestPage = new File('test/$servedFile.html');
+  await htmlTestPage.writeAsString(htmlTestPageContent);
+
+  RegExp path = new RegExp(r'(.*\/)');
+  File sauceRunnerFile = new File(
+      '${path.firstMatch(fullPathToTestFile).group(1)}sauce_runner.dart');
+  await sauceRunnerFile.writeAsString(sauceRunnerFileContent);
+
+  await Process.run('dart2js', [
+    path.firstMatch(fullPathToTestFile).group(1) + 'sauce_runner.dart',
+    '-o',
+    'test/$servedFile.js'
+  ]);
+
+  // platforms to run on sauce
+  List<List<String>> platforms = [
```

Comment:
pull these params out as configurable via dart_dev

even before that, I'd move this up to the top so that it's near the other configurable variables